theano/sandbox/cuda/basic_ops.py

```diff
@@ -3329,20 +3329,35 @@ def c_code_cache_version(self):
         return ()
 
 
-class GpuFlatten(gof.HideC, tensor.Flatten, GpuOp):
+class GpuFlatten(gof.HideC, tensor.Reshape, GpuOp):
     """
     Implement Flatten on the gpu.
 
     """
 
     def make_node(self, x):
+        warnings.warn(
+            "GpuFlatten class is deprecated, "
+            "please use gpu_flatten method instead.",
+            DeprecationWarning,
+            stacklevel=4)
         assert isinstance(x.type, CudaNdarrayType)
-        rval = tensor.Flatten.make_node(self, x)
+        rval = tensor.Reshape.make_node(self, x, [tensor.prod(x.shape)])
```

Comment:
No, ```GpuFlatten``` is deprecated as ```Flatten``` is. You should modify ```__init__.py``` to add a deprecation warning like in ```Flatten```.